grpc/server.go

```diff
@@ -59,14 +63,32 @@ func NewServer(
 	})
 
 	blockStreamService.SetPostHook(func(ctx context.Context, response *pbbstream.BlockResponseV2) {
-		//////////////////////////////////////////////////////////////////////
-		dmetering.EmitWithContext(dmetering.Event{
-			Source:      "firehose",
-			Kind:        "gRPC Stream",
-			Method:      "Blocks",
-			EgressBytes: int64(response.XXX_Size()),
-		}, ctx)
-		//////////////////////////////////////////////////////////////////////
+
+		block := &pbcodec.Block{}
+		err := ptypes.UnmarshalAny(response.Block, block)
+
+		if err != nil {
+			logger.Warn("failed to unmarshal block", zap.Error(err))
+		} else {
+
+			blockTime, err := block.Time()
+
+			// we slow down throughput if the allowed doc quota is not unlimited ("0") and unless it's live blocks (< 5 min)
+			if err == nil && time.Since(blockTime) > 5*time.Minute && rateLimit > 0 {
```

Comment:
So, is this to determine if block is "historical" in which case a rate limit should apply?